docs/proposals/conformance-test.md

```diff
@@ -0,0 +1,140 @@
+that we run for our e2e tests.
+
+This script will be converted to use Katib SDK. Tracking issue: https://github.com/kubeflow/katib/issues/2024.
+
+For the Training Operator, tests will be based on [the SDK e2e test.](https://github.com/kubeflow/training-operator/tree/05badc6ee8a071400efe9019d8d60fc242818589/sdk/python/test/e2e)
+
+### Test Workflow
+
+All tests will be run in the _kf-conformance_ namespace inside the separate container.
+That will help to avoid environment variance and improve fault tolerance. Driver is required to trigger the deployment and download the results.
+
+- We are going to use
+  [the unify Makefile](https://github.com/kubeflow/kubeflow/blob/2fa0d3665234125aeb8cebe8fe44f0a5a50791c5/conformance/1.5/Makefile)
+  for all Kubeflow conformance tests. Distributions (_driver_ on the diagram)
+  need to run the following Makefile commands:
+
+  ```makefile
+
+  # Run the conformance program.
+  run: setup run-katib run-training-operator
+
+  # Sets up the Kubernetes resources (Kubeflow Profile, RBAC) that needs to run the test.
+  # Create temporary folder for the conformance report.
+  setup:
+    kubectl apply -f ./setup.yaml
+    mkdir -p /tmp/kf-conformance
+
+  # Create deployment and run the e2e tests for Katib and Training Operator.
+  run-katib:
+    kubectl apply -f ./katib-conformance.yaml
+
+  run-training-operator:
+    kubectl apply -f ./training-operator-conformance.yaml
+
+  # Download the test deployment results to create PR for the Kubeflow Conformance Committee.
+  report:
+    ./report-conformance.sh
+
+  # Cleans up created resources and directories.
+  cleanup:
+    kubectl delete -f ./setup.yaml
+    kubectl delete -f ./katib-conformance.yaml
+    kubectl delete -f ./training-operator-conformance.yaml
+    rm -rf /tmp/kf-conformance
+  ```
+
+- Katib and Training Operator conformance deployment will have the appropriate
+  RBAC to Create/Read/Delete Katib Experiment and Training Operator Jobs in the
+  _kf-conformance_ namespace.
+
+- Distribution should have access to the internet to download the training datasets
+  (e.g. MNIST) while running the tests.
+
+- When the job is finished, the script generates output.
+
+  For Katib Experiment the output should be as follows:
+
+  ```
+  Test 1 - passed.
+  Experiment name: random-search
+  Experiment status: Experiment has succeeded because max trial count has reached
+  ```
+
+  For Training Operator the output should be as follows:
+
+  ```
+  Test 1 - passed.
+  TFJob name: tfjob-mnist
+  TFJob status: TFJob tfjob-mnist is successfully completed.
+  ```
+
+- The above report can be downloaded from the test deployment by running `make report`.
+
+- When all reports have been collected, the distributions are going to create PR
+  to publish the reports. The Kubeflow Conformance Committee will verify it and
```

Comment:
to publish the reports and to update the appropriate Kubeflow.org web pages on conformant Kubeflow distributions.